lib/macaroon.js

```diff
@@ -0,0 +1,206 @@
+import { importMacaroon, base64ToBytes } from 'macaroon'
+import { MacaroonId } from './macaroon-id'
+import isEqual from 'lodash/isEqual'
+import isEqualWith from 'lodash/isEqualWith'
+
+export const B64_REGEX = /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/
+export const HEX_REGEX = /^[0-9a-fA-F]+$/
+
+function decodeMacaroon (macaroon) {
+  if (HEX_REGEX.test(macaroon)) {
+    return importMacaroon(Buffer.from(macaroon, 'hex'))
+  }
+
+  if (B64_REGEX.test(macaroon)) {
+    return importMacaroon(Buffer.from(macaroon, 'base64'))
+  }
+
+  throw new Error('invalid macaroon encoding')
+}
+
+function macaroonOPs (macaroon) {
+  try {
+    const m = decodeMacaroon(macaroon)
+    const macJson = m.exportJSON()
+
+    if (macJson.i64) {
+      const identBytes = Buffer.from(base64ToBytes(macJson.i64))
+      if (identBytes[0] === 0x03) {
```

Comment:
Magic number with no comment

Reply:
From here: https://github.com/guggero/cryptography-toolkit/blob/2563355386943ba17e483ff53c8c8acf84d804bd/pages/macaroon/macaroon.js#L70

Should've included it